app/controllers/membership_applications_controller.rb

```diff
@@ -30,19 +19,27 @@ def edit
     @business_categories = BusinessCategory.all
   end
 
+  def create
+    @membership_application = current_user.membership_applications.new(membership_application_params)
+    if @membership_application.save
+      flash[:notice] = 'Thank you, Your application has been submitted'
+      redirect_to root_path
+    else
+      flash[:alert] = 'A problem prevented the membership application to be created'
+      render :new
+    end
+  end
+
   def update
     if @membership_application.update(membership_application_params)
-      flash[:notice] = 'Membership Application
-                        successfully updated'
+      flash[:notice] = 'Membership Application successfully updated'
       render :show
     else
-      flash[:alert] = 'A problem prevented the membership
-                       application to be saved'
+      flash[:alert] = 'A problem prevented the membership application to be saved'
       redirect_to edit_membership_application_path(@membership_application)
```

Comment:
I mean at line 39 `redirect_to edit_membership_application_path(@membership_application)`.
See the first answer to this question: [http://stackoverflow.com/questions/7493767/are-redirect-to-and-render-exchangeable](url)

Also there is no `@membership_application = MembershipApplication.find(params[:id])` in the edit action so my guess is this will give an error. Can you have a second look maybe try to (update) enter invalid data into the form and submit?